src/react-apps/applications/runtime/src/features/instantiate/containers/PartySelection.tsx

```diff
@@ -182,12 +181,12 @@ function PartySelection(props: IPartySelectionProps) {
     if (!language || !language.party_selection) {
       return null;
     }
-    if (location.state !== undefined &&
-      (location.state as IRedirectReason) &&
-      (location.state as IRedirectReason) !== undefined) {
-        switch ((location.state as IRedirectReason).errorType) {
+    if ((match.params as IRedirectParams).errorCode !== null) {
+      try {
+        const errorCode: number = parseInt((match.params as IRedirectParams).errorCode.toString(), 10);
```

Comment:
Med tanke på linje 184, der du også caster, hva om du samler casten utenfor ifen også bruker du den variabelen videre istedenfor multiple casts?

Reply:
The `toString()` is unnecessary, so I removed it. But I will still need to have a try/catch for parsing it into an integer, because a string does not match the enum. And the value is taken straight out of the URL where the user can type whatever.

Reply:
Nevermind, since I said that it has a enum-type, I have to cast to string for TypeScript to be a happy little boy :)

Reply:
I was thinking more about casting it once outside the if on line 184. instead of casting it twice. i.e.:
```
const params: IRedirectParams = match.params as IRedirectParams;
if(params.errorCode !== null) {
    try {
      const errorCode: number = parseInt(params.errorCode.toString(), 10);
    }
}
```

Just for readability. :)

Reply:
Agree on the readability part. Changing it.

Reply:
Even though `match.params as IRedirectParams;` is not casting in that sense. `parseInt(...)` and `.toString()` is casting. 

TypeScript is a developer-tool, that adds a layer of strong types on JavaScript. When we build and transpile the application, none of the TypeScript or interfaces we define is in the code. So TypeScript is essentially a tool that reduces the bugs we might create when sending in the wrong kind of data.